src/Simplic.OxS.ModelDefinition.Service/ModelDefinitionService.cs

```diff
@@ -0,0 +1,365 @@
+                        AddRefereceDefinition(modelDefinition, availableTypeAttribute.Type);
+                    }
+                }
+                if (property.PropertyType.IsSimpleType())
+                {
+                    (propertyDefinition.MinValue, propertyDefinition.MaxValue) = GetMinAndMaxValue(property);
+
+                    propertyDefinition.Type = Nullable.GetUnderlyingType(property.PropertyType)?.Name ?? property.PropertyType.Name;
+
+                    propertyDefinition.Format = GetFormat(property);
+                }
+                else if (property.PropertyType.IsCollectionType())
+                {
+                    Type? arrayType = null;
+
+                    if (property.PropertyType.IsArray)
+                        arrayType = property.PropertyType.GetElementType();
+
+                    if (property.PropertyType.IsGenericType)
+                        arrayType = property.PropertyType.GetGenericArguments()[0];
+
+                    if (arrayType is null && property.PropertyType.GetInterfaces().Any())
+                    {
+                        foreach (var interfaceType in property.GetType().GetInterfaces())
+                        {
+                            if (interfaceType.IsGenericType && interfaceType.GetGenericTypeDefinition() == typeof(IEnumerable<>))
+                            {
+                                arrayType = interfaceType.GetGenericArguments()[0];
+                                break;
+                            }
+                        }
+                    }
+
+                    if (arrayType is not null)
+                    {
+                        if (arrayType.IsSimpleType())
+                        {
+                            propertyDefinition.ArrayType = arrayType.ToString();
+                        }
+                        else
+                        {
+                            propertyDefinition.ArrayType = GetReferenceName(arrayType);
+                            AddRefereceDefinition(modelDefinition, arrayType);
+                        }
+                    }
+                }
+                else if (property.PropertyType.IsEnum)
+                {
+                    propertyDefinition.Type = property.PropertyType.Name;
+                    propertyDefinition.EnumType = Enum.GetUnderlyingType(property.PropertyType)?.ToString();
+                    propertyDefinition.EnumItems = GetEnumItems(property.GetType());
+                }
+                else
+                {
+                    propertyDefinition.Type = GetReferenceName(property.PropertyType);
+                    AddRefereceDefinition(modelDefinition, property.PropertyType);
+                }
+
+                properties.Add(propertyDefinition);
+            }
+
+            return properties;
+        }
+
+        private static (string?, string?) GetMinAndMaxValue(PropertyInfo property)
```

Comment:
I would suggest working with named tuples:

Like here with min and max: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples#use-cases-of-tuples

```suggestion
        private static (string? min, string? max) GetMinAndMaxValue(PropertyInfo property)
```